pkg/engine/vulnerability_builder_test.go

```diff
@@ -2,13 +2,234 @@ package engine
 
 import (
 	"fmt"
+	"path/filepath"
+	"runtime"
 	"testing"
 
 	"github.com/Checkmarx/kics/pkg/model"
 	"github.com/rs/zerolog"
 	"github.com/stretchr/testify/require"
 )
 
+type computeSimilarityIDParams struct {
+	filePath    string
+	queryID     string
+	searchKey   string
+	searchValue string
+}
+
+type computeSimilarityIDSubTest struct {
+	calls            []computeSimilarityIDParams
+	expectedFunction func(t *testing.T, firstHash, secondHash *string)
+}
+
+var (
+	similarityIDTests = []computeSimilarityIDSubTest{
+		{
+			calls: []computeSimilarityIDParams{
+				{
+					filePath:    "test.yaml",
+					queryID:     "e96ccbb0-8d74-49ef-87f8-b9613b63b6a8",
+					searchKey:   "Resources.MySearchKeyExample",
+					searchValue: "TCP,22",
+				},
+				{
+					filePath:    "test1.yaml", // change
+					queryID:     "e96ccbb0-8d74-49ef-87f8-b9613b63b6a8",
+					searchKey:   "Resources.MySearchKeyExample",
+					searchValue: "TCP,22",
+				},
+			},
+			expectedFunction: func(t *testing.T, firstHash, secondHash *string) {
+				require.NotEqual(t, *firstHash, *secondHash)
+			},
+		},
```

Comment:
Just to understand this part, why use this type of testing struct instead of splitting this test cases into different functions for each specific test?

Reply:
This blog post makes a case for table-driven tests: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

The main idea is to test all boundary conditions with minimum code duplication.

Reply:
Understood, seems good